homeassistant/components/packages.py

```diff
@@ -0,0 +1,76 @@
+"""
+Support for Packages.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/packages/
+"""
+import logging
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'packages'
+CONF_PACKAGES = DOMAIN
+
+_COMPONENT_SCHEMA = vol.Schema({cv.slug: vol.Any(dict, list)})
+
+_CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({cv.slug: _COMPONENT_SCHEMA})
```

Comment:
Package names and components in packages should be slugs. We dont want to combine the ways in which we can break up config for simplicity sake. For that reason I picked `cv.slug` and not [cv.slugify](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/helpers/config_validation.py#L229)